dist/ramda.js

```diff
@@ -4440,6 +4440,43 @@
         }, [], fns);
     };
 
+    // The algorithm used to handle cyclic structures is heavily
+    // inspired by underscore.js' implementation of isEqual
+    var _eqDeep = function _eqDeep(a, b, stackA, stackB) {
+        var idx = stackA.length;
+        while (--idx >= 0) {
+            if (stackA[idx] === a) {
+                return stackB[idx] === b;
+            }
+        }
+        stackA.push(a);
+        stackB.push(b);
+        var result;
+        switch (type(a)) {
+        case 'Array':
+        case 'Object':
+            if (keys(a).length !== keys(b).length || type(a) !== type(b)) {
+                result = false;
+            } else {
+                var keysA = keys(a);
```

Comment:
@davidrusu you shouldn't be calling `keys(a)` twice

That statement I made is slightly inaccurate as it won't handle a value on `a` being undefined while some other value on `b` is set (for instance `eq({a: undefined}, {b: 1})` :/